# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsPIDOMWindow.h
# Commit: 36b0646e8649
# Full Hash: 36b0646e86498662e770756ad54b1fd648a82184
# Author: Andreas Farre <farre@mozilla.com>
# Date: 2020-04-07 21:44:02
# Regressor Bug: 1620594
# File Overlap Count: 1
# Description:
#   Bug 1620594 - Part 7: Remove TabGroup and SystemGroup. r=nika,bas
#   
#   TabGroup never really made any difference in which thread something go
#   dispatched to. This was the intended use, but development of TabGroups
#   with abstract main threads never made it that far. The good thing is
# ==============================================================================

diff -r a5ad3b179c92 -r 36b0646e8649 dom/base/nsPIDOMWindow.h
--- a/dom/base/nsPIDOMWindow.h	Tue Apr 07 15:17:14 2020 +0000
+++ b/dom/base/nsPIDOMWindow.h	Tue Apr 07 15:17:47 2020 +0000
@@ -57,7 +57,6 @@
 class ContentFrameMessageManager;
 class DocGroup;
 class Document;
-class TabGroup;
 class Element;
 class Navigator;
 class Performance;
@@ -314,7 +313,7 @@
   void TryToCacheTopInnerWindow();
 
   // Increase/Decrease the number of active IndexedDB databases for the
-  // decision making of TabGroup scheduling and timeout-throttling.
+  // decision making of timeout-throttling.
   void UpdateActiveIndexedDBDatabaseCount(int32_t aDelta);
 
   // Return true if there is any active IndexedDB databases which could block
@@ -340,12 +339,6 @@
 
   void NoteDOMContentLoaded();
 
-  mozilla::dom::TabGroup* TabGroup();
-
-  // Like MaybeTabGroup, but it is more tolerant of being called at peculiar
-  // times, and it can return null.
-  mozilla::dom::TabGroup* MaybeTabGroup();
-
   virtual mozilla::dom::CustomElementRegistry* CustomElements() = 0;
 
   // XXX: This is called on inner windows
@@ -639,8 +632,6 @@
   // The AudioContexts created for the current document, if any.
   nsTArray<mozilla::dom::AudioContext*> mAudioContexts;  // Weak
 
-  RefPtr<mozilla::dom::TabGroup> mTabGroup;
-
   RefPtr<mozilla::dom::BrowsingContext> mBrowsingContext;
 
   // A unique (as long as our 64-bit counter doesn't roll over) id for
@@ -769,12 +760,6 @@
   bool IsTopLevelWindow();
   bool HadOriginalOpener() const;
 
-  mozilla::dom::TabGroup* TabGroup();
-
-  // Like MaybeTabGroup, but it is more tolerant of being called at peculiar
-  // times, and it can return null.
-  mozilla::dom::TabGroup* MaybeTabGroup();
-
   virtual nsPIDOMWindowOuter* GetPrivateRoot() = 0;
 
   virtual void ActivateOrDeactivate(bool aActivate) = 0;
@@ -876,6 +861,11 @@
   inline mozilla::dom::BrowsingContext* GetBrowsingContext() const;
 
   /**
+   * Get the browsing context group this window belongs to.
+   */
+  mozilla::dom::BrowsingContextGroup* GetBrowsingContextGroup() const;
+
+  /**
    * Set a new document in the window. Calling this method will in most cases
    * create a new inner window. This may be called with a pointer to the current
    * document, in that case the document remains unchanged, but a new inner
@@ -1139,8 +1129,6 @@
   // And these are the references between inner and outer windows.
   nsPIDOMWindowInner* MOZ_NON_OWNING_REF mInnerWindow;
 
-  RefPtr<mozilla::dom::TabGroup> mTabGroup;
-
   // A unique (as long as our 64-bit counter doesn't roll over) id for
   // this window.
   uint64_t mWindowID;